Sources/BijectiveDictionary/Documentation.docc/Creating a BijectiveDictionary.md

```diff
@@ -0,0 +1,69 @@
+# Creating a BijectiveDictionary
+
+Learn the many different ways to initialize a `BijectiveDictionary`.
+
+## Creating an Empty `BijectiveDictionary`
+An empty dictionary can be created using ``BijectiveDictionary/init()``
+Because `BijectiveDictionary` is generic, you will also need to somewhere declare your `Left` and `Right` value types. 
+
+```swift
+let bDict = BijectiveDictionary<String, Int>()
+```
+
+Or an empty `BijectiveDictionary` can also be created with a literal. However, this will require that you explicitly declare the type (unless it can be inferred elsewhere). 
+
+```swift
+let bDict: BijectiveDictionary<String, Int> = [:]
+```
+
+### Reserving Capacity
+An empty dictionary can also be created with a minimum capacity using ``BijectiveDictionary/init(minimumCapacity:)``. Remember that, (just like a standard `Dictionary`) when a `BijectiveDictionary` exceeds it's capacity, it must reallocate its storage buffer under the hood. This reallocation is automatic and you do not need to remember to do it, but it is not free. It is good to avoid this work, if you know in advance that you will have a large number of items. 
+
+```swift
+let bDict = BijectiveDictionary<String, Int>(minimumCapacity: 10_000)
+```
+
+## Initialize with Values
+### Create a `BijectiveDictionary` from a Literal
+See: ``BijectiveDictionary/init(dictionaryLiteral:)``
+
+``BijectiveDictionary`` conforms to `ExpressibleByDictionaryLiteral` and therefore can be initialized with common `Dictionary` syntax. (Note, this creates a `BijectiveDictionary` directly, and does not create any `Dictionary`.) In practice, this is one of the easiest ways to create a `BijectiveDictionary`.  
+
+># Warning
+>The dictionary literal used here must not contain any duplicates in the left or right values or else this initializer will fatal error. 
+```swift
+let bDict: BijectiveDictionary<String, Int> = ["A": 1, "B": 2, "C": 3]
```

Comment:
FYI, I tried the following syntax, but it looks like Swift is not able to infer Left and Right unfortunately. So we have to explicitly declare the type, including the generics. 

`let bDict: BijectiveDictionary = ["A": 1]

Reply:
That's interesting; the compiler is able to infer Left and Right when using this syntax in the tests. For instance:
https://github.com/ladvoc/BijectiveDictionary/blob/0521c33ca362ed8797ee13fccd6db5bb57d576b7/Tests/BijectiveDictionaryTests/BijectiveDictionaryTests.swift#L56

Reply:
I'm not sure why. It's possible that the following statements gave the compiler enough information to infer.